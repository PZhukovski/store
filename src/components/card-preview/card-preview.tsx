import React from "react";
import { Typography } from "@alfalab/core-components/typography";
import { Gap } from "@alfalab/core-components/gap";
import { ProductPreviewType } from "../../types/products";

import styles from "./cardpreview.module.css";

type MadeInAlfaPropsType = {
  product: ProductPreviewType;
};

export const CardPreview = ({ product }: MadeInAlfaPropsType) => {
  const { preview, title, price } = product;

  return (
    <div className={styles.block}>
      <div className={styles.wrapper_image}>
        <div
          className={styles.image}
          style={{ backgroundImage: `url(${preview})` }}></div>
      </div>
      <Gap size="m" />
      <Typography.TitleResponsive view="small" font="styrene" tag="h4">
        {title}
      </Typography.TitleResponsive>
      <Gap size="m" />
      <Typography.TitleResponsive
        view="small"
        font="styrene"
        weight="bold"
        tag="h4">
        {price} â‚½
      </Typography.TitleResponsive>
    </div>
  );
};
