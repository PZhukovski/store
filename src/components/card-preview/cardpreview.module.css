.block {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  margin-bottom: 30px;
}
.wrapper_image {
  position: relative;
  width: 100%;
  padding-bottom: 100%;
}
.image {
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 2;
}
