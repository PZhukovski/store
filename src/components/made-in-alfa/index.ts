export * from "./made-in-alfa"