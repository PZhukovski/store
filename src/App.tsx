import React, { useEffect } from "react";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import { useTypeSelector } from "../src/hooks/use-type-selector";
import { Page } from "./components/page";
import { Main } from "./components/main";
import { YourDesign } from "./components/your-design";
import { Card } from "./components/card"
import { MadeInAlfa } from "./components/made-in-alfa";
import { Contacts } from "./components/contacts";
import { ErrorBoundary } from "./error-boundary";
import { ButtonCart } from "./components/button-cart";
import { OrderType } from "./types/products";
import { Cart } from "./components/cart";
import { addLocalStorage } from "./slices/cart-slice";
import { useDispatch } from "react-redux";
var localStorage = require('store')

function App() {

  const products = useTypeSelector((state: { cart: { cart: OrderType[]; }; }) => state.cart.cart);
  const totalSum = useTypeSelector((state) => state.cart.totalSum);
  const dispatch = useDispatch();
  let cart =  localStorage.get('USER_CART')

  useEffect(() => {
    if (products.length === 0 &&  cart != undefined && cart != []) {
     dispatch(addLocalStorage(cart))
    }
   }, []);

   useEffect(() => {
    if (totalSum === 0 &&  cart != undefined && cart != []){
      localStorage.remove('USER_CART')
    }
  }, [totalSum]);

  return (
    <>
    <ErrorBoundary>
      <Router>
        <div className="App">
        { products.length > 0 &&
        <ButtonCart/>
        }
          <Routes>
            <Route path="/" element={<Page children={<Main/>}/>} />
            <Route path="/contact-us" element={<Page children={<Contacts/>}/>} />
            <Route path="/sdelano-v-alfe" element={<Page children={<MadeInAlfa/>}/>} />
            <Route path="/sdelano-v-alfe/:id" element={<Page children={<Card/>}/>} />
            <Route path="/svoy-dizain" element={<Page children={<YourDesign/>}/>} />
            <Route path="/svoy-dizain/:id" element={<Page children={<Card/>}/>}/>
            <Route path="/cart" element={<Page children={<Cart/>}/>} />
            <Route path="/policy" element ={''}/>
          </Routes>
        </div>
      </Router>
      </ErrorBoundary>
    </>
  );
}

export default App;
