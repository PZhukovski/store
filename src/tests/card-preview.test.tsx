import React from 'react';
import { render, screen } from '@testing-library/react';
import App from '../App';
import { CardPreview } from '../components/card-preview';
import '@testing-library/jest-dom/extend-expect';

test('snapshot', async() => {
  const cardElement = render(<CardPreview product={{
    id: 0,
    preview: '',
    title: '',
    price: 0,
    availability: false
  }}/>);
  expect(cardElement).toMatchSnapshot();
});

