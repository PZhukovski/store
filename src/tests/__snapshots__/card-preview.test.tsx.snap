// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="block"
      >
        <div
          class="wrapper_image"
        >
          <div
            class="image"
            style="background-image: url();"
          />
        </div>
        <div
          class="gap__gap_15oeq gap__vertical_15oeq"
          data-gap-size="m"
        />
        <h4
          class="typography__component_mub95 typography__styrene-small_g9g7d typography__medium_mub95"
        />
        <div
          class="gap__gap_15oeq gap__vertical_15oeq"
          data-gap-size="m"
        />
        <h4
          class="typography__component_mub95 typography__styrene-small_g9g7d typography__bold_mub95"
        >
          0
           ₽
        </h4>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="block"
    >
      <div
        class="wrapper_image"
      >
        <div
          class="image"
          style="background-image: url();"
        />
      </div>
      <div
        class="gap__gap_15oeq gap__vertical_15oeq"
        data-gap-size="m"
      />
      <h4
        class="typography__component_mub95 typography__styrene-small_g9g7d typography__medium_mub95"
      />
      <div
        class="gap__gap_15oeq gap__vertical_15oeq"
        data-gap-size="m"
      />
      <h4
        class="typography__component_mub95 typography__styrene-small_g9g7d typography__bold_mub95"
      >
        0
         ₽
      </h4>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
